lib/products.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+
+class ProductController extends StatefulWidget {
+  String msg = '';
+  bool value = false;
+  TextEditingController _controller;
```

Comment:
zarówno statefulWidget jak i statelessWidget powinny być immutable, dlatego warto dodać tutaj keyword final; msg niepotrzebne.
```suggestion
  final bool value = false;
  final TextEditingController _controller;
```